client/my-sites/themes/theme-options.js

```diff
@@ -188,12 +198,21 @@ export const connectOptions = connect(
 		return mapValues( ALL_THEME_OPTIONS, option => Object.assign(
 			{},
 			option,
+			( config.isEnabled( 'jetpack/pijp' ) && ( 'function' === typeof option.label ) )
+				? { label: ( option.label )( state, siteId ) }
+				: { label: option.label },
+			( config.isEnabled( 'jetpack/pijp' ) && ( 'function' === typeof option.extendedLabel ) )
+				? { extendedLabel: ( option.extendedLabel )( state, siteId ) }
+				: { extendedLabel: option.extendedLabel },
```

Comment:
Instead of adding this extra complexity (which would make `theme-options` even harder to grok 😛 ), I'd suggest adding a separate `upgradePlan` action -- since that's distinct enough from `purchase` -- and use `hideForTheme` to hide them where appropriate.